### PR TITLE
Transitive SDK Dependencies

### PR DESCRIPTION
**Pull request by adamkhazi**
Resolves https://github.com/tuist/tuist/issues/438

### Short description 📝

SDK dependencies for static frameworks or libraries were not previously being linked transitively.

### Solution 📦

Traverse the target node graph for the SDK dependencies in a similar way that we do already for static libraries and frameworks.

### Implementation 👩‍💻👨‍💻

- [X ] Update `TuistGenerator` to traverse graph recursively for SDK dependencies 
- [X ] Update `Graph+TestData` to support SDKs
- [X ] Add unit tests to check that transitive dependencies are added without duplicates for static libraries
- [X ] Update `ios_app_with_sdk` fixture to include transitive SDK dependency cases similar to the one failing in the issue raised
- [X ] Add acceptance test

### Test Plan

- Run `tuist generate` within `fixtures/ios_app_with_sdk`
- Verify that StaticFrameworkTests within the StaticFramework project has `libc++.tbd`
- Verify that StaticFrameworkTests builds and the tests pass

### CODE REVIEW

File: Tests/TuistGeneratorTests/Graph/GraphTests.swift [hunk 1]
```diff
@@ -457,3 +599,8 @@ final class DependencyReferenceTests: XCTestCase {
         XCTAssertFalse(DependencyReference.absolute("/B") < .product(target: "/A"))
     }
 }
+
+struct SDKPathAndStatus: Equatable {
```
Comment (pepibumur) [inline Tests/TuistGeneratorTests/Graph/GraphTests.swift]: Would `SDK` as a name conflict with other elements? What about `SystemSDK`?
